pysradb/sradb.py

```diff
@@ -1265,11 +1262,166 @@ def _srapath_url_srr(srr):
+            potentially other metadata pertaining to the accession numbers.
+        url_column: str
+            name of the dataframe column header that contains download
+            urls, or a regex matching the expected column header.
+            Defaults to None, in which case the regex
+            ".*sra.*(url|ftp|galaxy).*" will be used.
+
+        Returns
+        -------
+        df_for_download: Pandas.DataFrame
+            dataframe containing accession numbers and (preferably) URL links
+            that can be used for download.
+        """
+        missing_columns = []
+        df_columns = df.columns.tolist()
+        accession_columns = ["study_accession", "experiment_accession", "run_accession"]
+        for accession_column in accession_columns:
+            if accession_column not in df_columns:
+                missing_columns.append(accession_column)
+        # Special case for SraSearch
+        run_count = 1  # each row in the df contains at most 1 run_accession
+        while f"run_{run_count}_accession" in df_columns:
+            run_count += 1
+        if missing_columns == ["run_accession"] and run_count > 1:
+            missing_columns.clear()
+            accession_columns[-1] = "run_1_accession"
+        if url_column:
+            if url_column in df_columns:
+
+                formatted_df = df.loc[
+                    :, df.columns.isin(accession_columns + [url_column])
+                ]
+                formatted_df.rename(
+                    {"run_1_accession": "run_accession", url_column: "srapath_url"},
+                    inplace=True,
+                )
+            else:
+                formatted_df = df.loc[:, df.columns.isin(accession_columns)]
+                print(
+                    f"The URL column: {url_column} is not found.\n"
+                    "Generating default download URL for each run accession...\n",
+                    flush=True,
+                )
+        else:
+            run_dfs = []
+            url_regex = re.compile(".*sra.*(url|ftp|galaxy).*", re.IGNORECASE)
+            matched_cols = list(filter(url_regex.match, df_columns))
+            for i in range(1, run_count):
+                url_list = list(
+                    filter(lambda x: x.startswith(f"run_{i}"), matched_cols)
+                )
+                df["srapath_url"] = df.apply(
+                    lambda row: self._select_best_url(url_list, row), axis=1
+                )
+                expected_columns = [
+                    "study_accession",
+                    "experiment_accession",
+                    f"run_{i}_accession",
+                    "srapath_url",
+                ]
+                run_df = df.loc[:, df.columns.isin(expected_columns)]
+                run_df = run_df.rename(columns={f"run_{i}_accession": "run_accession"})
+                run_dfs.append(run_df)
+            if run_count == 1:
+                df["srapath_url"] = df.apply(
```

Comment:
We don't want to select the best url for the user. I think we should output all the URLS and let the user decide what url to choose.  If the user supplies the `--use_ascp` flag we use the aspera ftp downloads. See this notebook for an example of the possible use cases of pysradb download: https://github.com/saketkc/pysradb/blob/master/notebooks/08.pysradb_ascp_multithreaded.ipynb. The download function might need to be changed accordingly.

Reply:
Currently, download is implemented such that the user simply selects yes or no after seeing the list of all URLs. How should the user select which URL column to use?

Reply:
`download` accepts a [url_col](https://github.com/saketkc/pysradb/blob/5e54c4e37ab15f8f7ac3bc523238c486558770e2/pysradb/sradb.py#L1272) which is also exposed to the cLI. Maybe just use that?

Reply:
I think you are changing it to None in your code.

Reply:
Resolved by commits f1678dc to 0d3af0f